src/bootstrap/src/core/config/config.rs

```diff
@@ -2394,6 +2395,25 @@ impl Config {
         }
     }
 
+    fn is_compiler_changed(&self) -> bool {
+        let compiler_path = format!("{}", self.src.join("compiler/").display());
+        let library_path = format!("{}", self.src.join("library/").display());
```

Comment:
We should not be converting paths to String like this. You can pass a Path directly to an argument for some command. (Related recent thread: https://internals.rust-lang.org/t/i-think-the-name-path-display-should-be-replaced-with-path-display-lossy/20713/4)